### ISSUE TITLE
error:pitch difference too high to set automatic custo (z0)

### ISSUE THREAD

**Issue by jperon** [gregorio-project/gregorio#446]
I encounter this error on a real case : the [*Univérsi*](http://gregobase.selapa.net/chant.php?id=1747) gradual. When changing from `f3` to `c4` with `z0::c4`, gregorio throws this error. A possible workaround (inspired by GregoBase) would be `f+!//::c4`, but if this could be automatically computed, this would be better.

**Comment by rpspringuel** [comment]
This is a wrap-around issue:

Gregorio assumes that the FA of the `f3` clef is the FA above the DO of the `c4` clef.  This puts the RE custos that ends the `f3` section outside the range of the `c4` section (it would be well below the staff under this assumption).   The marking in the book indicates, however, that the FA of the `f3` clef should be considered the FA below the DO of the `c4` clef.  This puts the RE custos in the `f` position (as shown in the example).

What appears to be needed, then, is a way to specify whether a clef change is introducing higher notes or lower ones.

In pictures:

Gregorio assumes the following mapping between `f3` and `c4`:

```
f3:                FA SO LA TI DO RE MI FA SO LA TI DO RE
c4: LA TI DO RE MI FA SO LA TI DO RE MI FA
```
Note the RE in the `d` position on the `c4` staff is below the available notes in the `f3` staff.

What's needed (in this case at least) is the following mapping:

```
f3: FA SO LA TI DO RE MI FA SO LA TI DO RE
c4:       LA TI DO RE MI FA SO LA TI DO RE MI FA
```

Here there is a correspondence on that `d` position RE which shows the custos to be printed at the end of the `f3` section.

One could also envision two other mappings, which are even more extreme (and thus far less likely to be really practical):

```
f3:                                     FA SO LA TI DO RE MI FA SO LA TI DO RE
c4: LA TI DO RE MI FA SO LA TI DO RE MI FA
```

```
f3: FA SO LA TI DO RE MI FA SO LA TI DO RE
c4:                            LA TI DO RE MI FA SO LA TI DO RE MI FA
```

I think both of the first two cases are valid and should be allowed.  What we need, then is a way to indicate in the gabc which is desired and then the appropriate code to do the calculations correctly.  I don't think the last to cases are as important to support, I just thought I'd point out those possibilities for completeness sake.

Other clef changes may have similar issues.

**Comment by henryso** [comment]
I would postulate that the manual custos *is* the way to simply override whatever Gregorio does by default.  That said, before throwing the error message back at the user, I would suggest that Gregorio try the "other octave" to see if it can be used.  If the user doesn't like what they get, they can always just use the manual custos.

**Comment by eroux** [comment]
I agree with @henryso

**Comment by henryso** [comment]
Fix is in #449, ready for review.
